<launch>
    <arg name="image_width" default="1920" />
    <arg name="image_height" default="1080" />
    <arg name="fps" default="30" />
    <arg name="camera_info_path" default="$(find usb_cam_cv)/config" />

    <!-- camera0 -->
    <group ns="/">
        <arg name="camera_id" default="0" /> <!-- name id -->
        <arg name="camera_index" default="0" /> <!-- /dev/video -->
        <arg name="topic_name" default="/camera$(arg camera_id)/image_raw" />
        
        <node pkg="usb_cam_cv" type="usb_cam_node.py" name="camera$(arg camera_id)_publisher" output="screen">
            <param name="image_width" value="$(arg image_width)" />
            <param name="image_height" value="$(arg image_height)" />
            <param name="fps" value="$(arg fps)" />

            <param name="topic_name" value="$(arg topic_name)" />
            <param name="camera_index" value="$(arg camera_index)" />
            <param name="camera_id" value="$(arg camera_id)" />
            <param name="camera_info_topic_name" value="/camera$(arg camera_id)/camera_info" />
            <param name="camera_info_path" value="$(arg camera_info_path)/camera$(arg camera_id)_params.yaml" />
        </node>
    </group>

 
    <!-- camera1 -->
    <group ns="/">
        <arg name="camera_id" default="1" /> <!-- name id -->
        <arg name="camera_index" default="4" /> <!-- /dev/video -->
        <arg name="topic_name" default="/camera$(arg camera_id)/image_raw" />
        
        <node pkg="usb_cam_cv" type="usb_cam_node.py" name="camera$(arg camera_id)_publisher" output="screen">
            <param name="image_width" value="$(arg image_width)" />
            <param name="image_height" value="$(arg image_height)" />
            <param name="fps" value="$(arg fps)" />

            <param name="topic_name" value="$(arg topic_name)" />
            <param name="camera_index" value="$(arg camera_index)" />
            <param name="camera_id" value="$(arg camera_id)" />
            <param name="camera_info_topic_name" value="/camera$(arg camera_id)/camera_info" />
            <param name="camera_info_path" value="$(arg camera_info_path)/camera$(arg camera_id)_params.yaml" />
        </node>
    </group>

    <!-- camera2 -->
    <group ns="/">
        <arg name="camera_id" default="2" /> <!-- name id -->
        <arg name="camera_index" default="2" /> <!-- /dev/video -->
        <arg name="topic_name" default="/camera$(arg camera_id)/image_raw" />
        
        <node pkg="usb_cam_cv" type="usb_cam_node.py" name="camera$(arg camera_id)_publisher" output="screen">
            <param name="image_width" value="$(arg image_width)" />
            <param name="image_height" value="$(arg image_height)" />
            <param name="fps" value="$(arg fps)" />

            <param name="topic_name" value="$(arg topic_name)" />
            <param name="camera_index" value="$(arg camera_index)" />
            <param name="camera_id" value="$(arg camera_id)" />
            <param name="camera_info_topic_name" value="/camera$(arg camera_id)/camera_info" />
            <param name="camera_info_path" value="$(arg camera_info_path)/camera$(arg camera_id)_params.yaml" />
        </node>
    </group>

    <!-- camera3 -->
    <group ns="/">
        <arg name="camera_id" default="3" /> <!-- name id -->
        <arg name="camera_index" default="6" /> <!-- /dev/video -->
        <arg name="topic_name" default="/camera$(arg camera_id)/image_raw" />
        
        <node pkg="usb_cam_cv" type="usb_cam_node.py" name="camera$(arg camera_id)_publisher" output="screen">
            <param name="image_width" value="$(arg image_width)" />
            <param name="image_height" value="$(arg image_height)" />
            <param name="fps" value="$(arg fps)" />

            <param name="topic_name" value="$(arg topic_name)" />
            <param name="camera_index" value="$(arg camera_index)" />
            <param name="camera_id" value="$(arg camera_id)" />
            <param name="camera_info_topic_name" value="/camera$(arg camera_id)/camera_info" />
            <param name="camera_info_path" value="$(arg camera_info_path)/camera$(arg camera_id)_params.yaml" />
        </node>
    </group>

</launch>